### PR TITLE
postgresql: enable more features

### PR DESCRIPTION
**Pull request by poscat0x04**
###### Description of changes
Enabled the following features:

- ldap auth
- pam auth
- llvm JIT

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
This might be the fourth(?) attempt at enabling the postgres JIT. Please see previous PRs #124804, #150801 and probably more. cc @andir @abbradar @RaitoBezarius @marsam

**Comment by RaitoBezarius** [comment]
> This might be the fourth(?) attempt at enabling the postgres JIT. Please see previous PRs #124804, #150801 and probably more. cc @andir @abbradar @RaitoBezarius @marsam

Thank you for the triage, I suggest we try to get #150801 in shape, @poscat0x04 if that's okay with you, this should take care of the JIT part of this PR.

Then, we can examine the rest separately.

I can already tell unfortunately this PR has troubles with the WAL receiver test, see https://github.com/NixOS/nixpkgs/pull/218934/checks?check_run_id=11676876625

**Comment by kirillrdy** [comment]
jit support has been merged as part of https://github.com/NixOS/nixpkgs/pull/221851

File: pkgs/servers/sql/postgresql/default.nix [hunk 1]
```diff
@@ -68,17 +72,19 @@ let
...
     configureFlags = [
+      "--with-pam"
+      "--with-ldap"
       "--with-openssl"
       "--with-libxml"
       "--with-icu"
       "--sysconfdir=/etc"
       "--libdir=$(lib)/lib"
       "--with-system-tzdata=${tzdata}/share/zoneinfo"
-      "--enable-debug"
```
Comment (mweinelt) [inline pkgs/servers/sql/postgresql/default.nix:77]: Removing that should come with a motivation.

**Pull request closed without merging**